relay-server/src/actors/project_cache.rs

```diff
@@ -467,9 +521,15 @@ impl ProjectCacheBroker {
 
     /// Sends the message to the buffer service to dequeue the envelopes.
     ///
-    /// All the found envelopes will be send back through the `buffer_tx` channel and dirrectly
+    /// All the found envelopes will be send back through the `buffer_tx` channel and directly
     /// forwarded to `handle_processing`.
     pub fn dequeue(&mut self, partial_key: ProjectKey) {
+        // If we don't yet have the global config, we will defer dequeuing until we do.
+        if let GlobalConfigStatus::Pending(keys) = &mut self.global_config {
+            keys.insert(partial_key);
+            return;
+        }
```

Comment:
As I mentioned before, this potentially can be harmful , we can potentially accumulate a lot of data, in case we never can get the proper global config from the upstream.

Reply:
I had that in mind when making the recent change as a set of projectkeys should take a lot less space than a vector of messages 

i dont think the memory should be a problem here though, we already have many growing vectors in the projectcachebroker, although maybe we can use the indexed keys to fetch new states when the global config arrives. 

we could sync on this in case we end up going back-and-forth a lot here